### PR TITLE
Fix #222 - Email validation doesn't check for uniqueness

### LINKED ISSUE #222: Email validation doesn't check for uniqueness
Following from #178, `/signup` should also check to see if the email you use is unique (because of current loginapi server restriction).

### PR DESCRIPTION
**Pull request by alicoding**
fixes #222

### CODE REVIEW

**Comment by alicoding** [comment]
@secretrobotron R?

**Comment by alicoding** [comment]
or @cadecairos R?

**Comment by secretrobotron** [comment]
Gotta fix up those tests, it seems.

**Comment by alicoding** [comment]
@secretrobotron done! :D

File: templates/components/form/form.jsx [hunk 1]
```diff
@@ -53,12 +53,13 @@ var Form = React.createClass({
...
       errorMessage: {},
       valid_username: false,
+      valid_email: false,
       passwordError: false
     };
   },
   setFormState: function(data) {
     this.setState({
-      valid_username: true,
+      ['valid_'+data.field]: true,
```
Comment (cadecairos) [inline templates/components/form/form.jsx]: nice. can you add some whitespace here?

File: templates/components/form/form.jsx [hunk 2]
```diff
@@ -73,10 +74,21 @@ var Form = React.createClass({
   dirty: function(id, origin) {
     return (err, valid) => {
       if(err) {
+        if(id === 'email') {
+          this.setState({
+            ['valid_'+id]: false,
```
Comment (cadecairos) [inline templates/components/form/form.jsx]: if we know id is email, why not just key it as 'valid_email'

File: templates/components/form/form.jsx [hunk 3]
```diff
@@ -73,10 +74,21 @@ var Form = React.createClass({
...
+            errorMessage: {
+              [id]: 'Please use a valid email address.'
```
Comment (cadecairos) [inline templates/components/form/form.jsx]: same deal here

**Comment by secretrobotron** [comment]
Looks good. One problem I see:

1. Go to /signup.
2. Type in username that already exists.
3. Tab to email box (error should appear in username box).
4. Type in email that already exists.
5. Tab to password box (error should appear in email box).

The error for username will disappear:

screenshot 2015-04-16 09 12 21

Then, I can end up with a form that looks like this that won't submit, but there are no errors :(:
screenshot 2015-04-16 09 14 42

File: test/web.js [hunk 4]
```diff
@@ -73,28 +73,30 @@ lab.experiment("OAuth", function() {
       };
 
       s.inject(request, function(response) {
-        Code.expect(response.statusCode).to.equal(500);
+        Code.expect(response.statusCode).to.equal(400);
+        Code.expect(response.result.message).to.equal("Password not strong enough.");
```
Comment (cadecairos) [inline test/web.js]: give this test case a valid password, change it back to expecting a 500

File: test/web.js [hunk 5]
```diff
@@ -73,28 +73,30 @@ lab.experiment("OAuth", function() {
...
       Code.expect(error).to.be.undefined();
       var request = {
         method: "POST",
         url: "/create-user",
         payload: {
-          email: "webmaker@example.com",
+          email: "webmaker",
           username: "notgonnawork",
-          password: "CantGuessThis",
+          password: "CantGuessThis1234",
```
Comment (cadecairos) [inline test/web.js]: keep this valid password, fix the email param and title back to what they were, and then undo the changes to the assertions.

File: web/server.js [hunk 6]
```diff
@@ -260,6 +260,33 @@ module.exports = function(options) {
...
+                var err = Boom.badRequest('Password not strong enough.', result);
+                err.output.payload.details = err.data;
+                return reply(err);
+              }
+
+              reply(password);
+            }
+          }
+        ],
         validate: {
```
Comment (cadecairos) [inline web/server.js]: move this above the pre section.

File: web/server.js [hunk 7]
```diff
@@ -260,6 +260,33 @@ module.exports = function(options) {
...
+            assign: 'username',
+            method: function(request, reply) {
+              reply(request.payload.username);
+            }
+          },
+          {
+            assign: 'password',
+            method: function(request, reply) {
+              var password = request.payload.password;
+              if ( !password ) {
```
Comment (cadecairos) [inline web/server.js]: we don't need this check, since we're guaranteed it'll be here by the payload validation config
